backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java

```diff
@@ -210,6 +177,47 @@ public void getAllDatasetVersionsByDatasetId(
     }
   }
 
+  private DatasetVersionDTO getDatasetVersionDTOByDatasetId(
+      String datasetId, int pageNumber, int pageLimit)
+      throws InvalidProtocolBufferException, ModelDBException {
+    // Validate if current user has access to the entity or not
+    roleService.validateEntityUserWithUserInfo(
+        ModelDBServiceResourceTypes.REPOSITORY, datasetId, ModelDBServiceActions.READ);
+
+    /*Get Data*/
+    RepositoryIdentification repositoryIdentification =
+        RepositoryIdentification.newBuilder().setRepoId(Long.parseLong(datasetId)).build();
+    ListCommitsRequest.Builder listCommitsRequest =
+        ListCommitsRequest.newBuilder().setRepositoryId(repositoryIdentification);
+    if (pageLimit > 0 && pageNumber > 0) {
+      Pagination pagination =
+          Pagination.newBuilder().setPageLimit(pageLimit).setPageNumber(pageNumber).build();
+      listCommitsRequest.setPagination(pagination);
+    }
+    ListCommitsRequest.Response listCommitsResponse =
+        commitDAO.listCommits(
+            listCommitsRequest.build(),
+            (session -> repositoryDAO.getRepositoryById(session, repositoryIdentification)));
+
+    List<DatasetVersion> datasetVersions = new ArrayList<>();
+    long totalRecords = listCommitsResponse.getTotalRecords();
+    totalRecords = totalRecords > 0 ? totalRecords - 1 : totalRecords;
+
+    RepositoryEntity repositoryEntity = repositoryDAO.getRepositoryById(repositoryIdentification);
+    for (Commit commit : listCommitsResponse.getCommitsList()) {
+      if (commit.getMessage().equals(ModelDBConstants.INITIAL_COMMIT_MESSAGE)) {
```

Comment:
if commit does not have a parent is a better check than checking commit message

Reply:
Fixed